app/src/main/java/de/westnordost/streetcomplete/data/osmnotes/notequests/OsmNoteQuestController.kt

```diff
@@ -55,6 +57,15 @@ class OsmNoteQuestController(
         }
     }
 
+    private val spatialCache = SpatialCache(
+        SPATIAL_CACHE_SIZE,
+        16,
+        { bbox -> getAllVisibleInBBoxForCache(bbox) },
+        { keys -> noteCache.keys.removeAll(keys.toSet()) }
+    )
+    private val noteCache = HashMap<Long, OsmNoteQuest>()
```

Comment:
It seems inconsistent to have a cache for the `OsmNoteQuestController`. Unlike the `OsmQuestController`, it does not actually manage any persisted `noteQuestDB` but just generates the quests on demand from the OSM notes.

The other caches are all directly in reference to a database table, this one not.

Would it not make more sense to have a cache at `NoteController` (like there is a cache for `MapDataController`)? After all, for example the `OsmQuestController` also needs to access the notes (for blacklisting its positions) all the time, so that would be faster too if the cache was in `NoteController`. It's also easier in that class, because `NotController` is quite close to the DAO with little additional logic to take into account (e.g. `OsmQuestController` is quite a mess because of all this additional logic)

Reply:
I didn't notice the access to `NoteController` in `OsmQuestController`.
I agree, having the cache in `NoteController` does make sense.